u-boot/include/956x.h

```diff
@@ -0,0 +1,4502 @@
+/*
+ * QCA 956x series processor SOC registers
+ *
+ * (C) Copyright 2008 Atheros Communications, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ * MA 02111-1307 USA
+ */
+
+#ifndef _QCA956X_H
+#define _QCA956X_H
+
+#ifndef __ASSEMBLY__
+#include <asm/mipsregs.h>
+#include <asm/addrspace.h>
+#include <asm/types.h>
+#include <linux/types.h>
+#endif /* __ASSEMBLY__ */
+
+#undef is_qca956x
+#undef is_sco
+
```

Comment:
change to is_df for 963x

Reply:
These headers are a mess u-boot/include/atheros.h defines it, u-boot/include/956x.h redefine it. Result depend on headers order.

is_qca956x is used as a function returning a 0 or 1,  #ifdef is_qca956x is always true!
It must be undef before a redefine.

Reply:
they also define an alias name for chipset, in qca code it is a bug - sco=scorpion aka 955x. 956x is dragonfly. not sure these alias names are even used inside code actually...